### PR TITLE
Fix use of keyword args for intrinsic functions

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/3053
Towards: https://github.com/lfortran/lfortran/issues/3052

With this PR, I'm handling the presence of keywords with the args in intrinsic functions - https://github.com/lfortran/lfortran/issues/3053#issuecomment-1880593493
This should help in resolving the issue while building `stdlib`, however there are still changes that need to be done.

### CODE REVIEW

File: tests/reference/asr-array_size_02-94fb4d3.stderr [hunk 1]
```diff
@@ -1,5 +0,0 @@
-semantic error: dim has already been specified as a positional/keyword argument to size.
- --> tests/errors/array_size_02.f90:5:14
-  |
-5 |     size_a = size(a, 1, dim=1)
-  |              ^^^^^^^^^^^^^^^^^
```
Comment (HarshitaKalani) [inline tests/reference/asr-array_size_02-94fb4d3.stderr:5]: This error should be there.

**Comment by certik** [comment]
Thanks for working on this, very helpful.

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -4196,7 +4196,7 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
 
     ASR::asr_t* create_ArraySize(const AST::FuncCallOrArray_t& x) {
         Vec<ASR::expr_t*> args;
-        std::vector<std::string> kwarg_names = {"dim", "kind"};
+        std::vector<std::string> kwarg_names = {"array", "dim", "kind"};
         handle_intrinsic_node_args(x, args, kwarg_names, 1, 3, std::string("size"));
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:4200]: Let's separate these changes into another PR. But would modify the `handle_intrinsic_node_args` function to rename `kwarg_names` to just `arg_names` which would list all arguments, and then the min/max arg number would still be `1, 3` in this case, and everything should work.

Comment (certik) [reply src/lfortran/semantics/ast_common_visitor.h:4200]: Even later we can then refactor these to use the model in https://github.com/lfortran/lfortran/issues/3090.

Comment (certik) [reply src/lfortran/semantics/ast_common_visitor.h:4200]: For example the function `handle_intrinsic_node_args` can just access the `name2signature` dictionary to access the information.

**Comment by HarshitaKalani** [comment]
@certik this is ready. Can you please review it?

**Review by certik — APPROVED** [review]
I think this looks good, thanks!
